relayer/chains/cosmos/tx.go

```diff
@@ -485,17 +491,64 @@ func (cc *CosmosProvider) waitForBlockInclusion(
 }
 
 // mkTxResult decodes a comet transaction into an SDK TxResponse.
-func (cc *CosmosProvider) mkTxResult(resTx *coretypes.ResultTx) (*sdk.TxResponse, error) {
+func (cc *CosmosProvider) mkTxResult(resTx *client2.TxResponse) (*sdk.TxResponse, error) {
 	txbz, err := cc.Cdc.TxConfig.TxDecoder()(resTx.Tx)
 	if err != nil {
 		return nil, err
 	}
+
 	p, ok := txbz.(intoAny)
 	if !ok {
 		return nil, fmt.Errorf("expecting a type implementing intoAny, got: %T", txbz)
 	}
+
+	events := make([]abci.Event, len(resTx.ExecTx.Events))
```

Comment:
Would be nice to extract this logic to a method within the relayer `client` package